text/0046-metadata-for-offline-signers.md

```diff
@@ -0,0 +1,274 @@
+
+### Transmission
+
+Shortened metadata chunks MAY be trasmitted into cold device together with Merkle proof in its entirety or in parts, depending on memory capabilities of the cold device and it ability to reconstruct larger fraction of tree. This document does not specify the manner of transmission. The order of metadata chunks MAY be arbitrary, the only requirement is that indices of leaf nodes in Merkle tree corresponding to chunks MUST be communicated. Community MAY handle proof format standartization independently.
+
+### Offline verification
+
+The transmitted metadata chunks are hashed together with proof lemmas to obtain root that MAY be transmitted along with the rest of payload. Verification that the root transmitted with message matches with calculated root is optional; the transmitted root SHOULD NOT be used in signature, calculated root MUST be used; however, there is no mechanism to enforce this - it should be done during cold signers code audit.
+
+### Chain verification
+
+The root of metadata computed by cold device MAY be included into Signed Extensions; this way the transaction will pass as valid iff hash of metadata as seen by cold storage device is identical to consensus hash of metadata, ensuring fair signing protocol.
+
+The Signed Extension representing metadata digest is a single byte representing both digest vaule inclusion and shortening protocol version; this MUST be included in Signed Extensions set. Depending on its value, a digest value is included as `additionalSigned` to signature computation according to following specification:
+
+| signed extension value | digest value   | comment                            |
+|------------------------|----------------|------------------------------------|
+| `0x00`                 |                | digest is not included             |
+| `0x01`                 | 32-byte digest | this represents protocol version 1 |
+| `0x02` - `0xFF`        | *reserved*     | reserved for future use            |
+
+## Drawbacks
+
+### Increased transaction size
+
+A 1-byte increase in transaction size due to signed extension value. Digest is not included in transferred transaction, only in signing process.
+
+### Transition overhead
+
+Some slightly out of spec systems might experience breaking changes as new content of signed extensions is added. It is important to note, that there is no real overhead in processing time nor complexity, as the metadata checking mechanism is voluntary. The only drawbacks are expected for tools that do not implement MetadataV14 self-descripting features.
```

Comment:
But this is not related to the metadata version. Even if you know how to decode all the signed extensions using the type info of the metadata, you don't know how to construct the signed extension. It makes no sense that any tool would just pass `0` there, it should reject because it doesn't know the signed extension, which pjs is doing.

Reply:
```suggestion
Some slightly out of spec systems might experience breaking changes as new content of signed extensions is added - tools that delay their transition instead of preparing ahead of time would break for the duration of delay. It is important to note, that there is no real overhead in processing time nor complexity, as the metadata checking mechanism is voluntary.
```